contracts/vesting/src/contract.rs

```diff
@@ -0,0 +1,451 @@
+    if let Some(account) = load_account(&owner, deps.storage)? {
+        account.try_unbond_mixnode()
+    } else {
+        Err(ContractError::NoAccountForAddress(
+            owner.as_str().to_string(),
+        ))
+    }
+}
+
+pub fn try_track_unbond(
+    owner: Addr,
+    amount: Coin,
+    deps: DepsMut,
+) -> Result<Response, ContractError> {
+    let owner = deps.api.addr_validate(owner.as_str())?;
+    if let Some(account) = load_account(&owner, deps.storage)? {
+        account.track_unbond(amount, deps.storage)?;
+        Ok(Response::default())
+    } else {
+        Err(ContractError::NoAccountForAddress(
+            owner.as_str().to_string(),
+        ))
+    }
+}
+
+pub fn try_withdraw_vested_coins(
+    amount: Coin,
+    env: Env,
+    info: MessageInfo,
+    deps: DepsMut,
+) -> Result<Response, ContractError> {
+    let address = info.sender;
+    if let Some(account) = load_account(&address, deps.storage)? {
+        let spendable_coins = account.spendable_coins(None, &env, deps.storage)?;
+        if amount.amount < spendable_coins.amount {
+            let new_balance = account.load_balance(deps.storage)?.u128().saturating_sub(amount.amount.u128());
+            account.save_balance(Uint128::new(new_balance), deps.storage)?;
+
+            let send_tokens = BankMsg::Send {
+                to_address: address.as_str().to_string(),
+                amount: vec![amount],
+            };
+
+            Ok(Response::new()
+                .add_attribute("action", "whitdraw")
+                .add_message(send_tokens))
+        } else {
+            Err(ContractError::InsufficientSpendable(
+                address.as_str().to_string(),
+                spendable_coins.amount.u128(),
+            ))
+        }
+    } else {
+        return Err(ContractError::NoAccountForAddress(
+            address.as_str().to_string(),
+        ));
+    }
+}
+
+fn try_track_undelegation(
+    address: Addr,
+    mix_identity: IdentityKey,
+    amount: Coin,
+    info: MessageInfo,
+    deps: DepsMut,
+) -> Result<Response, ContractError> {
+    if info.sender != DEFAULT_MIXNET_CONTRACT_ADDRESS {
+        return Err(ContractError::NotMixnetContract(info.sender));
+    }
+    let address = deps.api.addr_validate(address.as_str())?;
+    if let Some(account) = load_account(&address, deps.storage)? {
+        account.track_undelegation(mix_identity, amount, deps.storage)?;
+        Ok(Response::default())
+    } else {
+        Err(ContractError::NoAccountForAddress(
+            address.as_str().to_string(),
+        ))
+    }
+}
+
+fn try_delegate_to_mixnode(
+    mix_identity: IdentityKey,
+    info: MessageInfo,
+    env: Env,
+    deps: DepsMut,
+) -> Result<Response, ContractError> {
+    let delegate_addr = info.sender;
+    let amount = validate_funds(&info.funds)?;
+    let address = deps.api.addr_validate(delegate_addr.as_str())?;
+    if let Some(account) = load_account(&address, deps.storage)? {
```

Comment:
All of the `if let Some(account) = ...` could be converted to `let account = ... .ok_or(err)?` and lose a level of imbrication (in other functions like `try_withdraw_vested_coins`, the advantage is even more visible)